chore: remove paginator dependency and use gapic to return streams for paginated calls instead

AVaksman opened the pull request:
gapic supports both paginated calls and its streamified versions. 

- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

AVaksman committed:
chore: drop paginator and use gapic to return streams for paginated calls instead

AVaksman committed:
chore: unit tests

stephenplusplus commented:
@AVaksman I know it's still a draft, so sorry if I'm jumping in too soon. But, I was wondering if you could describe the intentions behind this when you get the chance? Thanks!

AVaksman committed:
chore: more tests

AVaksman commented:
gapic now supports paginated calls and its streamified versions directly.

AVaksman committed:
Merge branch 'master' into drop_paginator

stephenplusplus commented:
Does it also retry requests?

AVaksman commented:
Yes, gax supports retries for paged calls.
`RetryOptions{retryCodes; backoffSettings;}` can be passed via `gaxOptions.retry` argument.
Streamified version of paged call is essentially same paged call where results pushed one by one via an object mode stream.

AVaksman committed:
refactor: better names

AVaksman committed:
refactor: minor

AVaksman commented on src/index.ts:
```diff
@@ -750,16 +750,20 @@ export class Bigtable {
       });
     };
 
+    const streamified = {listTablesStream: true};
```
Done

AVaksman commented on src/index.ts:
```diff
@@ -805,6 +809,20 @@ export class Bigtable {
       });
     }
+
+    function makeRequestStreamified() {
```
Done

AVaksman commented on src/instance.ts:
```diff
@@ -905,6 +908,71 @@ Please use the format 'my-instance' or '${bigtable.projectName}/instances/my-ins
+
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this;
+    const transform = function (
```
Done!
Have to pass tableId only thought, Table constructor checks for the proper projectId presence.

yoshi-automation committed:
chore: add config files for cloud-rad for node.js, delete Node 8 templates (#769)

This PR was generated using Autosynth. :rainbow:

Synth log will be available here:
https://source.cloud.google.com/results/invocations/5e903fff-57bb-4395-bb94-8b4d1909dbf6/targets

- [ ] To automatically regenerate this PR, check this box.

Source-Link: https://github.com/googleapis/synthtool/commit/21f1470ecd01424dc91c70f1a7c798e4e87d1eec
Source-Link: https://github.com/googleapis/synthtool/commit/388e10f5ae302d3e8de1fac99f3a95d1ab8f824a

yoshi-automation committed:
build: add Node 8 tests (#768)

This PR was generated using Autosynth. :rainbow:

Synth log will be available here:
https://source.cloud.google.com/results/invocations/5e903fff-57bb-4395-bb94-8b4d1909dbf6/targets

- [ ] To automatically regenerate this PR, check this box.

AVaksman committed:
refactor: fixy

AVaksman committed:
Merge branch 'master' into drop_paginator

stephenplusplus commented:
Thanks!

AVaksman committed:
chore: add error unit test

AVaksman committed:
Merge branch 'master' into drop_paginator

AVaksman committed:
Merge branch 'master' into drop_paginator

AVaksman committed:
Merge branch 'master' into drop_paginator

AVaksman commented:
@stephenplusplus I added another [unit test](https://github.com/googleapis/nodejs-bigtable/pull/762/commits/e9839fdfc8cfd9eaf12aaa343701215400c9af04) after the approval

stephenplusplus commented:
Nice, lgtm!
